src/chainlit/frontend/src/components/organisms/FormInput.tsx

```diff
@@ -0,0 +1,60 @@
+import Switch, { SwitchProps } from 'components/atoms/switch';
+
+import { IInput } from 'types/Input';
+
+import SelectInput, {
+  SelectInputProps,
+  SelectItem
+} from './inputs/selectInput';
+import TagsInput, { TagsInputProps } from './inputs/tagsInput';
+import TextInput, { TextInputProps } from './inputs/textInput';
+import Slider, { SliderProps } from './slider';
+
+interface IFormInput<T, I> extends IInput {
+  type: T;
+  initial: I;
+}
+
+export type TFormInput =
+  | (SwitchProps & IFormInput<'switch', boolean>)
+  | (SliderProps & IFormInput<'slider', number>)
+  | (SelectInputProps & IFormInput<'select', string>)
+  | (TextInputProps & IFormInput<'textinput', string>)
+  | (TagsInputProps & IFormInput<'tags', string>);
+
+const FormInput = ({ element }: { element: TFormInput }): JSX.Element => {
+  const renderInput = () => {
+    switch (element.type) {
+      case 'select':
+        return (
+          <SelectInput
+            {...element}
+            items={element?.items?.map((option: SelectItem) => ({
+              label: option.label,
+              value: option.value
+            }))}
+          />
+        );
+      case 'slider':
+        return <Slider {...element} />;
+      case 'tags':
+        return <TagsInput {...element} />;
+      case 'switch':
+        return (
+          <Switch
+            {...element}
+            inputProps={{
+              id: element.id || undefined,
+              name: element.id
+            }}
+          />
+        );
+      case 'textinput':
+        return <TextInput {...element} placeholder={element.placeholder} />;
+    }
+  };
+
+  return renderInput() || <></>;
```

Comment:
This should never happen and typescript is able to validate this if we add `"strictNullChecks": true,` in the `tsconfig.json` file.

Change the function to this:
```js
const FormInput = ({ element }: { element: TFormInput }): JSX.Element => {  
  switch (element.type) {
    case 'select':
      return (
        <SelectInput
          {...element}
          items={element.items?.map((option: SelectItem) => ({
            label: option.label,
            value: option.value
          }))}
        />
      );
    case 'slider':
      return <Slider {...element} />;
    case 'tags':
      return <TagsInput {...element} />;
    case 'switch':
      return (
        <Switch
          {...element}
          inputProps={{
            id: element.id || undefined,
            name: element.id
          }}
        />
      );
    case 'textinput':
      return <TextInput {...element} placeholder={element.placeholder} />;
  }
};
```

And now when you forget or add a new type, typescript will yell at you to implement the component :D

Reply:
But element should never ever be null, you can have a dynamic list of elements sure but allowing element null to be passed to the forminput is just not going to happen right?

Reply:
And if you add a default, typescript won't warn you anymore when you forget to implement a forminput

Reply:
We can't be sure that backend will send us a valid element type, what if backend accidentely send wrong type ? We can't trust typescript to fallback runtime check.

Reply:
And I don't see why typescript will not warn if the element is incorrect ? I've just had a fallback on the switch, the types are the same, it isn't null :)

Reply:
> 



> We can't be sure that backend will send us a valid element type, what if backend accidentely send wrong type ? We can't trust typescript to fallback runtime check.

Fair enough, I would imagine that is a major bug in the backend and normally I wouldn't consider this case but I can see why you want to do it. Would it be possible to add a typescript guard still like the following: https://medium.com/technogise/type-safe-and-exhaustive-switch-statements-aka-pattern-matching-in-typescript-e3febd433a7a

Reply:
We can discuss about it yes, why not.